### PR TITLE
add a basic machine

### PR DESCRIPTION
**Pull request by joelhooks**

### CODE REVIEW

File: src/components/EggheadPlayer/machine.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import {Machine} from 'xstate'
```
Comment (joelhooks) [inline src/components/EggheadPlayer/machine.ts:1]: xstate and typescript requires some love: https://xstate.js.org/docs/guides/typescript.html

File: src/pages/lessons/[slug].tsx [hunk 2]
```diff
@@ -109,6 +112,13 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
 
   if (!lesson) return null
 
+  React.useEffect(() => {
```
Comment (joelhooks) [inline src/pages/lessons/[slug].tsx:115]: if it isn't in a useEffect it cries in the console.

File: src/pages/lessons/[slug].tsx [hunk 3]
```diff
@@ -109,6 +112,13 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
...
+    if (playerState.value === 'loading') {
+      // here we can do some consideration for the actual state.
+      send('PLAY')
```
Comment (joelhooks) [inline src/pages/lessons/[slug].tsx:118]: I just defaulted to play here, but you can do all sorts of fun algorithms here to figure out the state based on data we have if it is pro or whatever to pop a CTA

File: src/pages/login.tsx [hunk 4]
```diff
@@ -11,33 +11,32 @@ const loginSchema = yup.object().shape({
 })
 
 function LoginForm() {
-  const [clicked, setClicked] = React.useState(false)
   const [isSubmitted, setIsSubmitted] = React.useState(false)
+  const [isError, setIsError] = React.useState(false)
```
Comment (joelhooks) [inline src/pages/login.tsx:15]: I fixed this on Epic React today and migrated it over so it just happened to be in this same PR.

File: src/pages/lessons/[slug].tsx (hunk 3 above)
Comment (johnlindquist) [reply src/pages/lessons/[slug].tsx:118]: Very cool. Should be a fun puzzle.

File: src/components/EggheadPlayer/machine.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+export type PlayerStateEvent =
+  | {type: 'PLAY'}
+  | {type: 'SUBSCRIBE'}
+  | {type: 'JOIN'}
+  | {type: 'QUIZ'}
+  | {type: 'COMPLETE'}
+  | {type: 'RATE'}
+
+const playerMachine = Machine<PlayerStateSchema, PlayerStateEvent>({
```
Comment (johnlindquist) [inline src/components/EggheadPlayer/machine.ts:23]: I bet you enjoyed writing this

File: src/pages/lessons/[slug].tsx (hunk 2 above)
Comment (johnlindquist) [reply src/pages/lessons/[slug].tsx:115]: I need to dive into the react/xstate/hooks demos

**Review by johnlindquist — APPROVED** [review]

**Pull request merged**